ansible_sdk/_util/metrics.py

```diff
@@ -0,0 +1,122 @@
+# Copyright: Ansible Project
+# Apache License 2.0 (see LICENSE or https://www.apache.org/licenses/LICENSE-2.0)
+
+import os
+import uuid
+
+from datetime import datetime
+
+import ansible_sdk._util.dataclass_compat as dataclasses
+from ansible_sdk.model.job_status import AnsibleJobStatus
+from ansible_sdk._aiocompat.csv_async import CSVAsync
+
+
+@dataclasses.dataclass(frozen=True, kw_only=True)
+class AnsiblePlaybookStats:
```

Comment:
I like your idea of using a dataclass to define the shape of the CSV- probably need to add a comment saying something like "don't mess with the order, it'll break the CSV column order" (though maybe it doesn't really matter? Depends on the consumer...).